### PR TITLE
field-unit-test  Added unit tests for Field

### PR DESCRIPTION
**Pull request by vigbk**

### CODE REVIEW

File: tests/core/field_test.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+from blurr.core.schema_loader import SchemaLoader
+
+@fixture
+def test_field_schema() -> Dict[str, Any]:
+    schema_loader = SchemaLoader()
+    name = schema_loader.add_schema(yaml.load('''
+Name: max_attempts
+Type: integer
+Value: 5
+'''))
```
Comment (prashastk) [inline tests/core/field_test.py]: instead of yaml load should we pass in a dict?

File: tests/core/field_test.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    assert field.snapshot == 0
+    
+    field.evaluate()
+    assert field.snapshot == 5
+
+
+def test_field_restore(test_field_schema):
+    field = MockField(test_field_schema, EvaluationContext())
+    field.restore(5)
+    assert field.value == 5
```
Comment (prashastk) [inline tests/core/field_test.py]: newline at end

File: tests/core/field_test.py (hunk 1 above)
Comment (vigbk) [reply tests/core/field_test.py]: sure. will change that.

File: tests/core/field_test.py (hunk 2 above)
Comment (vigbk) [reply tests/core/field_test.py]: Noted

**Pull request merged**